lib/opentelemetry_phoenix.ex

```diff
@@ -195,12 +195,21 @@ defmodule OpentelemetryPhoenix do
   end
 
   defp client_ip(%{remote_ip: remote_ip} = conn) do
-    case Plug.Conn.get_req_header(conn, "x-forwarded-for") do
-      [] ->
-        to_string(:inet_parse.ntoa(remote_ip))
+    case header_value(conn, "x-forwarded-for") |> parse_x_forwarded_for() do
+      {:ok, ip_address} ->
+        ip_address
+
+      _ ->
+        remote_ip |> :inet_parse.ntoa() |> to_string()
+    end
+  end
```

Comment:
The result of `header_value` is always a binary, so I'm not sure of why the guard is there on 207 and we add the extra steps here.

Assuming keeping the change to use `header_value`, could this just be like so?

```suggestion
defp client_ip(%{remote_ip: remote_ip} = conn) do
    case header_value(conn, "x-forwarded-for") do
      "" ->
        remote_ip
        |> :inet_parse.ntoa()
        |> to_string()
        
      ip_address ->
        ip_address
        |> String.split(",", parts: 2)
        |> List.first("")
    end
  end
end
```

Reply:
That's indeed a bit redundant, your suggestion makes more sense. However, do you also want to skip "validating" the IP address given the fact it comes from a user-editable header? The validation  was implicitly done in `parse_x_forwarded_for`.

Reply:
We don't really care; it's a string either way. :)

Gateways are supposed to strip that header on entry to prevent users from messing with it. Generally, we don't want to do any more parsing, validating, or manipulating than absolutely necessary.

Reply:
In that case, done 👍🏻